src/compiler/transformers/declarations/diagnostics.ts

```diff
@@ -767,7 +767,7 @@ export function createGetIsolatedDeclarationErrors(resolver: EmitResolver): (nod
         if (isSetAccessor(node.parent)) {
             return createAccessorTypeError(node.parent);
         }
-        const addUndefined = resolver.requiresAddingImplicitUndefined(node, /*enclosingDeclaration*/ undefined);
+        const addUndefined = resolver.requiresAddingImplicitUndefined(node, findAncestor(node, isFunctionLikeDeclaration));

```

Comment:
this is a somewhat cheap way of fixing this issue, given the importance of the `enclosingDeclaration` for the correct result `resolver.requiresAddingImplicitUndefined`, I really wonder if it shouldn't be made a required parameter

Reply:
It's already required, it's just that it can be undefined, you just don't really want to pass in undefined unless via some goofy callback, the checker gave it undefined.

Reply:
But, maybe external users should be required to provide it.